Run also tests with MPI

At the moment Travis only runs debug and release tests with clang. We should also run the mpi versions for both release and debug.

when I try to compile trilinos with mpi on  travis I get the following errors:

```
[4/1590] Building CXX object packages/...hreads/Kokkos_Threads_TaskPolicy.cpp.o
FAILED: /usr/bin/mpicxx   -Dkokkoscore_EXPORTS -pedantic -Wall -Wno-long-long -Wwrite-strings   -w -std=c++0x -O3 -fPIC -I. -Ipackages/kokkos/core/src -I../packages/kokkos/core/src -MMD -MT packages/kokkos/core/src/CMakeFiles/kokkoscore.dir/Threads/Kokkos_Threads_TaskPolicy.cpp.o -MF packages/kokkos/core/src/CMakeFiles/kokkoscore.dir/Threads/Kokkos_Threads_TaskPolicy.cpp.o.d -o packages/kokkos/core/src/CMakeFiles/kokkoscore.dir/Threads/Kokkos_Threads_TaskPolicy.cpp.o -c ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.cpp
In file included from ../packages/kokkos/core/src/Kokkos_Core_fwd.hpp:51:0,
                 from ../packages/kokkos/core/src/Kokkos_Threads.hpp:47,
                 from ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.hpp:50,
                 from ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.cpp:49:
../packages/kokkos/core/src/Kokkos_Macros.hpp:223:6: error: #error "Compiling with GCC version earlier than 4.7.2 is not supported."
In file included from ../packages/kokkos/core/src/Kokkos_HostSpace.hpp:61:0,
                 from ../packages/kokkos/core/src/Kokkos_Threads.hpp:53,
                 from ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.hpp:50,
                 from ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.cpp:49:
../packages/kokkos/core/src/impl/KokkosExp_SharedAlloc.hpp: In constructor ‘Kokkos::Experimental::Impl::SharedAllocationRecord<void, void>::SharedAllocationRecord()’:
../packages/kokkos/core/src/impl/KokkosExp_SharedAlloc.hpp:113:6: error: ‘this’ is not a potential constant expression
In file included from ../packages/kokkos/core/src/Kokkos_Threads.hpp:53:0,
                 from ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.hpp:50,
                 from ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.cpp:49:
../packages/kokkos/core/src/Kokkos_HostSpace.hpp: At global scope:
../packages/kokkos/core/src/Kokkos_HostSpace.hpp:196:3: error: a class-key must be used when declaring a friend
../packages/kokkos/core/src/Kokkos_HostSpace.hpp:196:3: error: friend declaration does not name a class or function
In file included from ../packages/kokkos/core/src/KokkosExp_View.hpp:330:0,
                 from ../packages/kokkos/core/src/Kokkos_View.hpp:2099,
                 from ../packages/kokkos/core/src/Kokkos_Parallel.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Threads.hpp:206,
                 from ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.hpp:50,
                 from ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.cpp:49:
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:711:9: error: expected nested-name-specifier before ‘is_mapping_plugin’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:711:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:711:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:711:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:723:9: error: expected nested-name-specifier before ‘is_mapping_plugin’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:723:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:723:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:723:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:724:9: error: expected nested-name-specifier before ‘is_regular’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:724:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:724:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:724:27: error: expected unqualified-id before ‘=’ token
In file included from ../packages/kokkos/core/src/KokkosExp_View.hpp:330:0,
                 from ../packages/kokkos/core/src/Kokkos_View.hpp:2099,
                 from ../packages/kokkos/core/src/Kokkos_Parallel.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Threads.hpp:206,
                 from ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.hpp:50,
                 from ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.cpp:49:
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:947:9: error: expected nested-name-specifier before ‘is_mapping_plugin’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:947:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:947:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:947:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:948:9: error: expected nested-name-specifier before ‘is_regular’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:948:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:948:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:948:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1196:9: error: expected nested-name-specifier before ‘is_mapping_plugin’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1196:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1196:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1196:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1197:9: error: expected nested-name-specifier before ‘is_regular’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1197:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1197:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1197:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1421:9: error: expected nested-name-specifier before ‘is_mapping_plugin’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1421:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1421:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1421:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1422:9: error: expected nested-name-specifier before ‘is_regular’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1422:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1422:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1422:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1821:9: error: expected nested-name-specifier before ‘is_mapping_plugin’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1821:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1821:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1821:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1822:9: error: expected nested-name-specifier before ‘is_regular’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1822:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1822:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1822:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp: In member function ‘const Kokkos::Experimental::Impl::ALL_t& Kokkos::Experimental::Impl::ALL_t::operator()() const’:
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:2143:63: error: ‘this’ is not a potential constant expression
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp: At global scope:
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:2355:9: error: expected nested-name-specifier before ‘is_regular’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:2355:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:2355:20: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:2355:20: error: expected unqualified-id before ‘=’ token
In file included from ../packages/kokkos/core/src/KokkosExp_View.hpp:332:0,
                 from ../packages/kokkos/core/src/Kokkos_View.hpp:2099,
                 from ../packages/kokkos/core/src/Kokkos_Parallel.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Threads.hpp:206,
                 from ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.hpp:50,
                 from ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.cpp:49:
../packages/kokkos/core/src/impl/KokkosExp_ViewArray.hpp:175:9: error: expected nested-name-specifier before ‘is_regular’
../packages/kokkos/core/src/impl/KokkosExp_ViewArray.hpp:175:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewArray.hpp:175:20: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewArray.hpp:175:20: error: expected unqualified-id before ‘=’ token
In file included from ../packages/kokkos/core/src/Kokkos_View.hpp:2099:0,
                 from ../packages/kokkos/core/src/Kokkos_Parallel.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Threads.hpp:206,
                 from ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.hpp:50,
                 from ../packages/kokkos/core/src/Threads/Kokkos_Threads_TaskPolicy.cpp:49:
../packages/kokkos/core/src/KokkosExp_View.hpp:383:1: error: expected unqualified-id before ‘using’
../packages/kokkos/core/src/KokkosExp_View.hpp:1073:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1093:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1112:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1130:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1146:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1160:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1173:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1184:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1203:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1247:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1289:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1329:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1366:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1400:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1432:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1461:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
[4/1590] Building CXX object packages/...e.dir/impl/KokkosExp_SharedAlloc.cpp.o
FAILED: /usr/bin/mpicxx   -Dkokkoscore_EXPORTS -pedantic -Wall -Wno-long-long -Wwrite-strings   -w -std=c++0x -O3 -fPIC -I. -Ipackages/kokkos/core/src -I../packages/kokkos/core/src -MMD -MT packages/kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/KokkosExp_SharedAlloc.cpp.o -MF packages/kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/KokkosExp_SharedAlloc.cpp.o.d -o packages/kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/KokkosExp_SharedAlloc.cpp.o -c ../packages/kokkos/core/src/impl/KokkosExp_SharedAlloc.cpp
In file included from ../packages/kokkos/core/src/Kokkos_Core_fwd.hpp:51:0,
                 from ../packages/kokkos/core/src/Kokkos_Core.hpp:50,
                 from ../packages/kokkos/core/src/impl/KokkosExp_SharedAlloc.cpp:44:
../packages/kokkos/core/src/Kokkos_Macros.hpp:223:6: error: #error "Compiling with GCC version earlier than 4.7.2 is not supported."
In file included from ../packages/kokkos/core/src/Kokkos_HostSpace.hpp:61:0,
                 from ../packages/kokkos/core/src/Kokkos_View.hpp:50,
                 from ../packages/kokkos/core/src/Kokkos_Parallel.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Serial.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Core.hpp:61,
                 from ../packages/kokkos/core/src/impl/KokkosExp_SharedAlloc.cpp:44:
../packages/kokkos/core/src/impl/KokkosExp_SharedAlloc.hpp: In constructor ‘Kokkos::Experimental::Impl::SharedAllocationRecord<void, void>::SharedAllocationRecord()’:
../packages/kokkos/core/src/impl/KokkosExp_SharedAlloc.hpp:113:6: error: ‘this’ is not a potential constant expression
In file included from ../packages/kokkos/core/src/Kokkos_View.hpp:50:0,
                 from ../packages/kokkos/core/src/Kokkos_Parallel.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Serial.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Core.hpp:61,
                 from ../packages/kokkos/core/src/impl/KokkosExp_SharedAlloc.cpp:44:
../packages/kokkos/core/src/Kokkos_HostSpace.hpp: At global scope:
../packages/kokkos/core/src/Kokkos_HostSpace.hpp:196:3: error: a class-key must be used when declaring a friend
../packages/kokkos/core/src/Kokkos_HostSpace.hpp:196:3: error: friend declaration does not name a class or function
In file included from ../packages/kokkos/core/src/KokkosExp_View.hpp:330:0,
                 from ../packages/kokkos/core/src/Kokkos_View.hpp:2099,
                 from ../packages/kokkos/core/src/Kokkos_Parallel.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Serial.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Core.hpp:61,
                 from ../packages/kokkos/core/src/impl/KokkosExp_SharedAlloc.cpp:44:
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:711:9: error: expected nested-name-specifier before ‘is_mapping_plugin’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:711:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:711:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:711:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:723:9: error: expected nested-name-specifier before ‘is_mapping_plugin’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:723:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:723:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:723:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:724:9: error: expected nested-name-specifier before ‘is_regular’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:724:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:724:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:724:27: error: expected unqualified-id before ‘=’ token
In file included from ../packages/kokkos/core/src/KokkosExp_View.hpp:330:0,
                 from ../packages/kokkos/core/src/Kokkos_View.hpp:2099,
                 from ../packages/kokkos/core/src/Kokkos_Parallel.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Serial.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Core.hpp:61,
                 from ../packages/kokkos/core/src/impl/KokkosExp_SharedAlloc.cpp:44:
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:947:9: error: expected nested-name-specifier before ‘is_mapping_plugin’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:947:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:947:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:947:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:948:9: error: expected nested-name-specifier before ‘is_regular’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:948:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:948:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:948:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1196:9: error: expected nested-name-specifier before ‘is_mapping_plugin’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1196:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1196:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1196:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1197:9: error: expected nested-name-specifier before ‘is_regular’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1197:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1197:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1197:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1421:9: error: expected nested-name-specifier before ‘is_mapping_plugin’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1421:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1421:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1421:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1422:9: error: expected nested-name-specifier before ‘is_regular’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1422:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1422:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1422:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1821:9: error: expected nested-name-specifier before ‘is_mapping_plugin’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1821:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1821:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1821:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1822:9: error: expected nested-name-specifier before ‘is_regular’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1822:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1822:27: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:1822:27: error: expected unqualified-id before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp: In member function ‘const Kokkos::Experimental::Impl::ALL_t& Kokkos::Experimental::Impl::ALL_t::operator()() const’:
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:2143:63: error: ‘this’ is not a potential constant expression
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp: At global scope:
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:2355:9: error: expected nested-name-specifier before ‘is_regular’
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:2355:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:2355:20: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewMapping.hpp:2355:20: error: expected unqualified-id before ‘=’ token
In file included from ../packages/kokkos/core/src/KokkosExp_View.hpp:332:0,
                 from ../packages/kokkos/core/src/Kokkos_View.hpp:2099,
                 from ../packages/kokkos/core/src/Kokkos_Parallel.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Serial.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Core.hpp:61,
                 from ../packages/kokkos/core/src/impl/KokkosExp_SharedAlloc.cpp:44:
../packages/kokkos/core/src/impl/KokkosExp_ViewArray.hpp:175:9: error: expected nested-name-specifier before ‘is_regular’
../packages/kokkos/core/src/impl/KokkosExp_ViewArray.hpp:175:9: error: using-declaration for non-member at class scope
../packages/kokkos/core/src/impl/KokkosExp_ViewArray.hpp:175:20: error: expected ‘;’ before ‘=’ token
../packages/kokkos/core/src/impl/KokkosExp_ViewArray.hpp:175:20: error: expected unqualified-id before ‘=’ token
In file included from ../packages/kokkos/core/src/Kokkos_View.hpp:2099:0,
                 from ../packages/kokkos/core/src/Kokkos_Parallel.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Serial.hpp:52,
                 from ../packages/kokkos/core/src/Kokkos_Core.hpp:61,
                 from ../packages/kokkos/core/src/impl/KokkosExp_SharedAlloc.cpp:44:
../packages/kokkos/core/src/KokkosExp_View.hpp:383:1: error: expected unqualified-id before ‘using’
../packages/kokkos/core/src/KokkosExp_View.hpp:1073:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1093:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1112:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1130:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1146:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1160:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1173:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1184:3: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1203:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1247:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1289:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1329:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1366:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1400:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1432:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
../packages/kokkos/core/src/KokkosExp_View.hpp:1461:1: error: ‘Subview’ in namespace ‘Kokkos::Experimental’ does not name a type
[4/1590] Building CXX object packages/...ir/impl/Kokkos_AllocationTracker.cpp.o
FAILED: /usr/bin/mpicxx   -Dkokkoscore_EXPORTS -pedantic -Wall -Wno-long-long -Wwrite-strings   -w -std=c++0x -O3 -fPIC -I. -Ipackages/kokkos/core/src -I../packages/kokkos/core/src -MMD -MT packages/kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_AllocationTracker.cpp.o -MF packages/kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_AllocationTracker.cpp.o.d -o packages/kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_AllocationTracker.cpp.o -c ../packages/kokkos/core/src/impl/Kokkos_AllocationTracker.cpp
In file included from ../packages/kokkos/core/src/Kokkos_Core_fwd.hpp:51:0,
                 from ../packages/kokkos/core/src/impl/Kokkos_AllocationTracker.cpp:44:
../packages/kokkos/core/src/Kokkos_Macros.hpp:223:6: error: #error "Compiling with GCC version earlier than 4.7.2 is not supported."
In file included from ../packages/kokkos/core/src/Kokkos_HostSpace.hpp:61:0,
                 from ../packages/kokkos/core/src/Kokkos_Atomic.hpp:71,
                 from ../packages/kokkos/core/src/impl/Kokkos_AllocationTracker.cpp:48:
../packages/kokkos/core/src/impl/KokkosExp_SharedAlloc.hpp: In constructor ‘Kokkos::Experimental::Impl::SharedAllocationRecord<void, void>::SharedAllocationRecord()’:
../packages/kokkos/core/src/impl/KokkosExp_SharedAlloc.hpp:113:6: error: ‘this’ is not a potential constant expression
In file included from ../packages/kokkos/core/src/Kokkos_Atomic.hpp:71:0,
                 from ../packages/kokkos/core/src/impl/Kokkos_AllocationTracker.cpp:48:
../packages/kokkos/core/src/Kokkos_HostSpace.hpp: At global scope:
../packages/kokkos/core/src/Kokkos_HostSpace.hpp:196:3: error: a class-key must be used when declaring a friend
../packages/kokkos/core/src/Kokkos_HostSpace.hpp:196:3: error: friend declaration does not name a class or function
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,53 +9,59 @@ cache:
 notifications:
   email: false
 
-addons:
+addons: &gcc5
   apt:
     sources:
-      - ubuntu-toolchain-r-test
-      - llvm-toolchain-precise-3.6
-      - george-edison55-precise-backports
+    - ubuntu-toolchain-r-test
     packages:
-      - ninja-build
-      - cmake
-      - cmake-data
-      - build-essential
-      - libblas-dev
-      - liblapack-dev
-      - libboost-dev
-      - clang-3.6
-      - gfortran
-      - doxygen
-      - graphviz
+    - build-essential
+    - libblas-dev
+    - liblapack-dev
+    - libboost-dev
+    - doxygen
+#    - graphviz
+    - g++-5
+    - gcc-5
+    - gfortran
 
+compiler: gcc-5
 language: cpp
-compiler: clang
 
 before_install:
+  - export PRG=$PWD/programs
+  - export CASA=$PWD
+  - export CXX=g++-5
+  - export CC=gcc-5
   - ./scripts/setup_astyle.sh
   - export PATH=$PWD/programs/astyle/build/gcc/bin:$PATH
   - ./scripts/check_indentation.sh
-
-install:
-  - export PRG=$PWD/programs
-  - export CASA=$PWD
-  - export CXX=clang++-3.6
-  - export PATH=$PWD/programs/cmake/bin:$PATH
-  - export PATH=$PWD/programs/ninja:$PATH
   - export TRILINOS_DIR=$PWD/programs/trilinos
   - export DEAL_II_DIR=$PWD/programs/dealii
-  - export SUNDIALS_DIR=$PWD/programs/sundials
+  - export DEAL2LKIT_DIR=$CASA/deal2lkit-$BUILD_TYPE
+  - export P4EST_DIR=$PWD/programs/p4est
   - export PATH=$PWD/programs/astyle/build/gcc/bin:$PATH
   - export PATH=$PRG/numdiff/bin:$PATH
-  - ./scripts/set_cache.sh
-  - mkdir build
-  - cd build
-  - cmake -GNinja -DCMAKE_INSTALL_PREFIX=/home/travis/deal2lkit-$BUILD_TYPE -DD2K_ENABLE_TESTING=ON -DCMAKE_BUILD_TYPE=$BUILD_TYPE ..
+  - ./scripts/travis/install_mpi.sh
+  - export PATH=$PRG/mpich/bin:$PATH
+  - ./scripts/travis/install_cmake.sh
+  - export PATH=$PRG/cmake/bin:$PATH
+  - ./scripts/travis/install_ninja.sh
+  - export PATH=$PRG/ninja:$PATH
+  - ./scripts/travis/install_numdiff.sh
+  - export CXX=$PRG/mpich/bin/mpicxx
+  - export CC=$PRG/mpich/bin/mpicc
+  - ./scripts/travis/install_p4est.sh
+  - ./scripts/travis/install_sundials.sh
+  - export SUNDIALS_DIR=$PRG/sundials
+
+install: true
+  
+script:
+  - ./scripts/travis/install_trilinos_OR_dealii.sh
+  - cmake -GNinja -DCMAKE_INSTALL_PREFIX=/home/travis/deal2lkit-$BUILD_TYPE -DD2K_ENABLE_TESTING=ON -DCMAKE_BUILD_TYPE=$BUILD_TYPE
   - ninja -j4 install
   - tar cfz /home/travis/deal2lkit-CI-$BUILD_TYPE.tgz /home/travis/deal2lkit-$BUILD_TYPE
   - tar cfz /home/travis/dealii-CI-environment.tgz $PRG
-  
-script:
   - ninja setup_tests
   - ctest -j4 -V -R "${BUILD_TYPE,,}"
     

--- a/include/deal2lkit/revision.h.in
+++ b/include/deal2lkit/revision.h.in
@@ -17,6 +17,8 @@
 #ifndef _d2k_revision_h
 #define _d2k_revision_h
 
+#include <deal2lkit/config.h>
+
 /**
  * Name of the local git branch of the source directory.
  */

--- a/scripts/travis/install_cmake.sh
+++ b/scripts/travis/install_cmake.sh
@@ -1,0 +1,17 @@
+#!/bin/sh
+
+PRG=$PWD/programs
+CASA=$PWD
+
+# installing cmake
+if [ ! -d $PRG/cmake ]
+then
+  echo "-------------------------------------->installing cmake"
+  wget --no-check-certificate https://cmake.org/files/v3.5/cmake-3.5.0-rc3-Linux-x86_64.tar.gz
+  tar xzf cmake-3.5.0-rc3-Linux-x86_64.tar.gz -C $PRG
+  rm cmake-3.5.0-rc3-Linux-x86_64.tar.gz
+  mv $PRG/cmake-* $PRG/cmake
+fi
+export PATH=$PRG/cmake/bin:$PATH
+
+

--- a/scripts/travis/install_dealii.sh
+++ b/scripts/travis/install_dealii.sh
@@ -1,0 +1,31 @@
+#!/bin/sh
+
+PRG=$PWD/programs
+CASA=$PWD
+
+
+# dealii
+if [ ! -d $PRG/dealii ]
+then
+  echo "-------------------------------------->installing dealii"
+  DST_INST=$PRG/dealii 
+  cd $PRG
+  git clone https://github.com/dealii/dealii.git dealii-tmp
+  cd dealii-tmp
+  mkdir build
+  cd build
+  cmake \
+    -G Ninja \
+    -D CMAKE_INSTALL_PREFIX:PATH=$DST_INST \
+    -D DEAL_II_WITH_MPI:BOOL=ON \
+    -D DEAL_II_WITH_THREADS:BOOL=OFF \
+    -D DEAL_II_WITH_P4EST:BOOL=ON \
+    .. > $CASA/dealii_cmake.log 2>&1
+  cat summary.log
+  ninja -j3 -k0 ; ninja -j3 -k0; ninja -j2 -k0 ; ninja -j2 -k0; ninja -j1 -k0 ; ninja -j1
+  ninja -j4 install > /dev/null
+  cd $PRG
+  rm -rf dealii-tmp
+  cd $CASA
+fi
+

--- a/scripts/travis/install_mpi.sh
+++ b/scripts/travis/install_mpi.sh
@@ -1,0 +1,23 @@
+#!/bin/sh
+
+PRG=$PWD/programs
+CASA=$PWD
+
+
+# mpich
+if [ ! -d $PRG/mpich ]
+then
+  echo "-------------------------------------->installing mpich"
+  mkdir $PRG/mpich-tmp
+  cd $PRG/mpich-tmp
+  wget --no-check-certificate http://www.mpich.org/static/downloads/3.2/mpich-3.2.tar.gz
+  tar xvzf mpich-3.2.tar.gz > /dev/null
+  mv mpich-3.2 mpich
+   cd mpich
+   mkdir build && cd build
+   ../configure CFLAGS="-w" --prefix=$PRG/mpich --disable-static > /dev/null
+   make -j4 install > /dev/null
+   cd $PRG
+   rm -rf $PRG/mpich-tmp
+fi
+export PATH=$PRG/mpich/bin:$PATH

--- a/scripts/travis/install_ninja.sh
+++ b/scripts/travis/install_ninja.sh
@@ -1,0 +1,20 @@
+#!/bin/sh
+
+PRG=$PWD/programs
+CASA=$PWD
+
+
+# installing ninja
+if [ ! -d $PRG/ninja ]
+then
+  echo "-------------------------------------->installing ninja"
+  cd $PRG
+  git clone git://github.com/martine/ninja.git
+  cd ninja
+  git checkout release
+  ./configure.py --bootstrap > /dev/null
+  cd $CASA
+fi
+export PATH=$PWD/programs/ninja:$PATH
+
+

--- a/scripts/travis/install_numdiff.sh
+++ b/scripts/travis/install_numdiff.sh
@@ -1,0 +1,25 @@
+#!/bin/sh
+
+PRG=$PWD/programs
+CASA=$PWD
+
+
+
+# installing numdiff
+if [ ! -d $PRG/numdiff ]
+then
+  echo "-------------------------------------->installing numdiff"
+  mkdir $PRG/numdiff-tmp
+  cd $PRG/numdiff-tmp
+  wget http://mirror.lihnidos.org/GNU/savannah/numdiff/numdiff-5.8.1.tar.gz
+  tar xzf numdiff-5.8.1.tar.gz
+  rm numdiff-5.8.1.tar.gz
+  cd numdiff-5.8.1
+  DST_INST=$PRG/numdiff
+  ./configure --prefix=$DST_INST > /dev/null
+  make -j4 install > /dev/null
+  cd $CASA
+  rm -rf $PRG/numdiff-tmp
+fi
+
+

--- a/scripts/travis/install_p4est.sh
+++ b/scripts/travis/install_p4est.sh
@@ -1,0 +1,26 @@
+#!/bin/sh
+
+PRG=$PWD/programs
+CASA=$PWD
+
+
+# p4est
+
+if [ ! -d $PRG/p4est ]
+then
+    echo "-------------------------------------->installing p4est"
+    DST_INST=$PRG/p4est
+    cd $PRG
+    git clone https://github.com/cburstedde/p4est.git p4est-tmp
+    cd p4est-tmp
+    git submodule init; git submodule update
+    ./bootstrap > /dev/null
+    ./configure --enable-mpi --prefix=$DST_INST > /dev/null
+    make -j4 > /dev/null
+    make install
+    cd $CASA
+    rm -rf $PRG/p4est-tmp
+fi
+export P4EST_DIR=$PRG/p4est
+    
+

--- a/scripts/travis/install_sundials.sh
+++ b/scripts/travis/install_sundials.sh
@@ -1,0 +1,36 @@
+#!/bin/sh
+
+PRG=$PWD/programs
+CASA=$PWD
+
+
+
+# sundials
+if [ ! -d $PRG/sundials ]
+then
+  echo "-------------------------------------->installing sundials"
+  DST_INST=$PRG/sundials
+  cd $PRG
+  wget http://pkgs.fedoraproject.org/repo/extras/sundials/sundials-2.6.2.tar.gz/3deeb0ede9f514184c6bd83ecab77d95/sundials-2.6.2.tar.gz
+  tar xzf sundials-2.6.2.tar.gz
+  rm sundials-2.6.2.tar.gz
+  mv sundials-2.6.2 sundials-tmp
+  cd sundials-tmp
+  mkdir build
+  cd build
+#  export PATH=$PRG/cmake/bin:$PATH
+  export PATH=$PWD/programs/ninja:$PATH
+  cmake \
+    -G Ninja \
+    -D CMAKE_INSTALL_PREFIX:PATH=$DST_INST \
+    -D CXX_ENABLE:BOOL=ON \
+    -D MPI_ENABLE:BOOL=ON \
+    -D MPI_CC:PATH=$PRG/mpich/bin/mpicc \
+    -D MPI_CXX:PATH=$PRG/mpich/bin/mpicxx \
+    ..
+  ninja -j4 install
+  cd $PRG
+  rm -rf sundials-tmp
+  cd $CASA
+fi
+export SUNDIALS_DIR=$PRG/sundials

--- a/scripts/travis/install_trilinos.sh
+++ b/scripts/travis/install_trilinos.sh
@@ -1,0 +1,64 @@
+#!/bin/sh
+
+PRG=$PWD/programs
+CASA=$PWD
+
+
+# trilinos
+if [ ! -d $PRG/trilinos ]
+then
+  echo "-------------------------------------->installing trilinos"
+  DST_INST=$PRG/trilinos
+  export PATH=$PRG/cmake/bin:$PATH
+  cd $PRG
+  git clone https://github.com/trilinos/trilinos.git trilinos-tmp
+  cd trilinos-tmp
+  mkdir build
+  cd build
+  cmake \
+    -G Ninja \
+    -D CMAKE_BUILD_TYPE:STRING=RELEASE \
+    -D TPL_ENABLE_Boost:BOOL=OFF \
+    -D TPL_ENABLE_BoostLib:BOOL=OFF \
+    -D TPL_ENABLE_MPI:BOOL=ON \
+    -D TPL_ENABLE_Netcdf:BOOL=OFF \
+    -D CMAKE_INSTALL_PREFIX:PATH=$DST_INST \
+    -D Trilinos_ENABLE_OpenMP:BOOL=OFF \
+    -D BUILD_SHARED_LIBS:BOOL=ON \
+    -D Trilinos_WARNINGS_AS_ERRORS_FLAGS:STRING="" \
+    -D CMAKE_VERBOSE_MAKEFILE:BOOL=FALSE \
+    -D Trilinos_ASSERT_MISSING_PACKAGES:BOOL=OFF \
+    -D Trilinos_ENABLE_TESTS:BOOL=OFF \
+    -D Trilinos_ENABLE_ALL_PACKAGES:BOOL=OFF \
+    -D Trilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=OFF \
+    -D Trilinos_ENABLE_Epetra:BOOL=ON \
+    -D Trilinos_ENABLE_NOX:BOOL=ON \
+    -D Trilinos_ENABLE_EpetraExt:BOOL=ON \
+    -D Trilinos_ENABLE_Tpetra:BOOL=ON \
+    -D Trilinos_ENABLE_Kokkos:BOOL=ON \
+    -D Trilinos_ENABLE_Sacado:BOOL=ON \
+    -D Trilinos_ENABLE_Amesos:BOOL=ON \
+    -D Trilinos_ENABLE_AztecOO:BOOL=ON \
+    -D Trilinos_ENABLE_Ifpack:BOOL=ON \
+    -D Trilinos_ENABLE_Rythmos:BOOL=ON \
+    -D Trilinos_ENABLE_Piro:BOOL=ON \
+    -D Trilinos_ENABLE_MOOCHO:BOOL=ON \
+    -D Trilinos_ENABLE_ML:BOOL=ON \
+    -D Trilinos_ENABLE_MueLu:BOOL=ON \
+    -D Trilinos_ENABLE_Komplex:BOOL=ON \
+    -D Trilinos_ENABLE_Thyra:BOOL=ON \
+    -D Trilinos_ENABLE_TrilinosCouplings:BOOL=ON \
+    -D Trilinos_ENABLE_Fortran:BOOL=OFF \
+    -D Trilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON \
+    -D CMAKE_CXX_COMPILER:PATH=$PRG/mpich/bin/mpicxx\
+    -D CMAKE_C_COMPILER:PATH=$PRG/mpich/bin/mpicc \
+    .. > $CASA/trilinos_cmake.log 2>&1
+
+  ninja -j3 
+  ninja -j4 install > /dev/null
+  cd $PRG
+  rm -rf trilinos-tmp
+  cd $CASA
+fi
+
+export TRILINOS_DIR=$PRG/trilinos

--- a/scripts/travis/install_trilinos.sh~
+++ b/scripts/travis/install_trilinos.sh~
@@ -1,0 +1,63 @@
+#!/bin/sh
+
+PRG=$PWD/programs
+CASA=$PWD
+
+
+# trilinos
+if [ ! -d $PRG/trilinos ]
+then
+  echo "-------------------------------------->installing trilinos"
+  DST_INST=$PRG/trilinos
+  export PATH=$PRG/cmake/bin:$PATH
+  cd $PRG
+  git clone https://github.com/trilinos/trilinos.git trilinos-tmp
+  cd trilinos-tmp
+  mkdir build
+  cd build
+  cmake \
+    -G Ninja \
+    -D CMAKE_BUILD_TYPE:STRING=RELEASE \
+    -D TPL_ENABLE_Boost:BOOL=OFF \
+    -D TPL_ENABLE_BoostLib:BOOL=OFF \
+    -D TPL_ENABLE_MPI:BOOL=ON \
+    -D TPL_ENABLE_Netcdf:BOOL=OFF \
+    -D CMAKE_INSTALL_PREFIX:PATH=$DST_INST \
+    -D Trilinos_ENABLE_OpenMP:BOOL=OFF \
+    -D BUILD_SHARED_LIBS:BOOL=ON \
+    -D Trilinos_WARNINGS_AS_ERRORS_FLAGS:STRING="" \
+    -D CMAKE_VERBOSE_MAKEFILE:BOOL=FALSE \
+    -D Trilinos_ASSERT_MISSING_PACKAGES:BOOL=OFF \
+    -D Trilinos_ENABLE_TESTS:BOOL=OFF \
+    -D Trilinos_ENABLE_ALL_PACKAGES:BOOL=OFF \
+    -D Trilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=OFF \
+    -D Trilinos_ENABLE_Epetra:BOOL=ON \
+    -D Trilinos_ENABLE_NOX:BOOL=ON \
+    -D Trilinos_ENABLE_EpetraExt:BOOL=ON \
+    -D Trilinos_ENABLE_Tpetra:BOOL=ON \
+    -D Trilinos_ENABLE_Kokkos:BOOL=ON \
+    -D Trilinos_ENABLE_Sacado:BOOL=ON \
+    -D Trilinos_ENABLE_Amesos:BOOL=ON \
+    -D Trilinos_ENABLE_AztecOO:BOOL=ON \
+    -D Trilinos_ENABLE_Ifpack:BOOL=ON \
+    -D Trilinos_ENABLE_Rythmos:BOOL=ON \
+    -D Trilinos_ENABLE_Piro:BOOL=ON \
+    -D Trilinos_ENABLE_MOOCHO:BOOL=ON \
+    -D Trilinos_ENABLE_ML:BOOL=ON \
+    -D Trilinos_ENABLE_MueLu:BOOL=ON \
+    -D Trilinos_ENABLE_Komplex:BOOL=ON \
+    -D Trilinos_ENABLE_Thyra:BOOL=ON \
+    -D Trilinos_ENABLE_TrilinosCouplings:BOOL=ON \
+    -D Trilinos_ENABLE_Fortran:BOOL=OFF \
+    -D CMAKE_CXX_COMPILER:PATH=$PRG/mpich/bin/mpicxx\
+    -D CMAKE_C_COMPILER:PATH=$PRG/mpich/bin/mpicc \
+    .. > $CASA/trilinos_cmake.log 2>&1
+
+  ninja -j3 
+  ninja -j4 install > /dev/null
+  cd $PRG
+  rm -rf trilinos-tmp
+  cd $CASA
+fi
+
+export TRILINOS_DIR=$PRG/trilinos

--- a/scripts/travis/install_trilinos_OR_dealii.sh
+++ b/scripts/travis/install_trilinos_OR_dealii.sh
@@ -1,0 +1,17 @@
+#!/bin/sh
+
+PRG=$PWD/programs
+CASA=$PWD
+
+
+# trilinos
+if [ ! -d $PRG/trilinos -a ! -d $PRG/dealii ]
+then
+    $CASA/scripts/travis/install_trilinos.sh
+else
+    if [ ! -d $PRG/dealii ]
+    then
+	$CASA/scripts/travis/install_dealii.sh
+    fi
+fi
+

--- a/scripts/travis/install_trilinos_and_dealii.sh~
+++ b/scripts/travis/install_trilinos_and_dealii.sh~
@@ -1,0 +1,63 @@
+#!/bin/sh
+
+PRG=$PWD/programs
+CASA=$PWD
+
+
+# trilinos
+if [ ! -d $PRG/trilinos ]
+then
+  echo "-------------------------------------->installing trilinos"
+  DST_INST=$PRG/trilinos
+  export PATH=$PRG/cmake/bin:$PATH
+  cd $PRG
+  git clone https://github.com/trilinos/trilinos.git trilinos-tmp
+  cd trilinos-tmp
+  mkdir build
+  cd build
+  cmake \
+    -G Ninja \
+    -D CMAKE_BUILD_TYPE:STRING=RELEASE \
+    -D TPL_ENABLE_Boost:BOOL=OFF \
+    -D TPL_ENABLE_BoostLib:BOOL=OFF \
+    -D TPL_ENABLE_MPI:BOOL=ON \
+    -D TPL_ENABLE_Netcdf:BOOL=OFF \
+    -D CMAKE_INSTALL_PREFIX:PATH=$DST_INST \
+    -D Trilinos_ENABLE_OpenMP:BOOL=OFF \
+    -D BUILD_SHARED_LIBS:BOOL=ON \
+    -D Trilinos_WARNINGS_AS_ERRORS_FLAGS:STRING="" \
+    -D CMAKE_VERBOSE_MAKEFILE:BOOL=FALSE \
+    -D Trilinos_ASSERT_MISSING_PACKAGES:BOOL=OFF \
+    -D Trilinos_ENABLE_TESTS:BOOL=OFF \
+    -D Trilinos_ENABLE_ALL_PACKAGES:BOOL=OFF \
+    -D Trilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=OFF \
+    -D Trilinos_ENABLE_Epetra:BOOL=ON \
+    -D Trilinos_ENABLE_NOX:BOOL=ON \
+    -D Trilinos_ENABLE_EpetraExt:BOOL=ON \
+    -D Trilinos_ENABLE_Tpetra:BOOL=ON \
+    -D Trilinos_ENABLE_Kokkos:BOOL=ON \
+    -D Trilinos_ENABLE_Sacado:BOOL=ON \
+    -D Trilinos_ENABLE_Amesos:BOOL=ON \
+    -D Trilinos_ENABLE_AztecOO:BOOL=ON \
+    -D Trilinos_ENABLE_Ifpack:BOOL=ON \
+    -D Trilinos_ENABLE_Rythmos:BOOL=ON \
+    -D Trilinos_ENABLE_Piro:BOOL=ON \
+    -D Trilinos_ENABLE_MOOCHO:BOOL=ON \
+    -D Trilinos_ENABLE_ML:BOOL=ON \
+    -D Trilinos_ENABLE_MueLu:BOOL=ON \
+    -D Trilinos_ENABLE_Komplex:BOOL=ON \
+    -D Trilinos_ENABLE_Thyra:BOOL=ON \
+    -D Trilinos_ENABLE_TrilinosCouplings:BOOL=ON \
+    -D Trilinos_ENABLE_Fortran:BOOL=OFF \
+    -D CMAKE_CXX_COMPILER:PATH=$PRG/mpich/bin/mpicxx\
+    -D CMAKE_C_COMPILER:PATH=$PRG/mpich/bin/mpicc \
+    .. > $CASA/trilinos_cmake.log 2>&1
+
+  ninja -j3 
+  ninja -j4 install > /dev/null
+  cd $PRG
+  rm -rf trilinos-tmp
+  cd $CASA
+fi
+
+export TRILINOS_DIR=$PRG/trilinos

